src/openrct2/rct2/T6Importer.cpp

```diff
@@ -200,8 +206,30 @@ class TD6Importer final : public ITrackImporter
         }
 
         td->name = _name;
+
+        UpdateRideType(td);
+
         return td;
     }
+
+    void UpdateRideType(std::unique_ptr<TrackDesign>& td)
+    {
+        if (RCT2RideTypeNeedsConversion(td->type))
+        {
+            std::scoped_lock<std::mutex> lock(_objectLookupMutex);
```

Comment:
Is this function actually used from multiple threads? The only things that use multiple threads is the index builder and the drawing code, this might not be needed here.

Reply:
Eh nevermind, I absolutely ignored I added this since this is in the file scanner.